osu.Game/Screens/Edit/EditorClock.cs

```diff
@@ -216,7 +222,21 @@ public void SeekSmoothlyTo(double seekDestination)
 
         public bool IsRunning => underlyingClock.IsRunning;
 
-        public void ProcessFrame() => underlyingClock.ProcessFrame();
+        public void ProcessFrame()
+        {
+            underlyingClock.ProcessFrame();
+
+            var playbackAlreadyStopped = playbackFinished;
+            playbackFinished = CurrentTime >= TrackLength;
+
+            if (playbackFinished && !playbackAlreadyStopped)
+            {
+                if (IsRunning)
+                    underlyingClock.Stop();
+
+                underlyingClock.Seek(TrackLength);
+            }
+        }
```

Comment:
I'm thinking that this could be simplified further to

```csharp
        public void ProcessFrame()
        {
            underlyingClock.ProcessFrame();

            playbackFinished = CurrentTime >= TrackLength;

            if (playbackFinished)
            {
                if (IsRunning)
                    underlyingClock.Stop();

                if (CurrentTime > TrackLength)
                    underlyingClock.Seek(TrackLength);
            }
        }
```

Reasoning for why:
* Gets rid of another conditional variable
* Ensures that the time always is rolled back to `TrackLength` (I can imagine a scenario where a clock seek stops at `TrackLength` first, but then, because in the current code we have `playbackAlreadyStopped`, you could technically seek further past the track length, as that variable being `true` makes the code inside dead).

Thoughts?

Reply:
Seems like a no-brainer to me